pkg/osutil/osutil.go

```diff
@@ -33,3 +37,50 @@ func Unsetenv(key string) error {
 	}
 	return nil
 }
+
+// InterruptHandler is a function that is called on receiving a
+// SIGTERM or SIGINT signal.
+type InterruptHandler func()
+
+var (
+	interruptMu sync.Mutex
+	// interruptHandlers holds all registered InterruptHandlers in order
+	// they will be executed.
+	interruptHandlers = []InterruptHandler{}
+)
+
+// RegisterInterruptHandler registers a new InterruptHandler.
+func RegisterInterruptHandler(h InterruptHandler) {
+	interruptMu.Lock()
+	defer interruptMu.Unlock()
+	interruptHandlers = append(interruptHandlers, h)
+}
+
+// HandleInterrupts calls the handler functions on receiving a SIGINT or SIGTERM.
+func HandleInterrupts() {
+	notifier := make(chan os.Signal, 1)
+	signal.Notify(notifier, syscall.SIGINT, syscall.SIGTERM)
+
+	go func() {
+		sig := <-notifier
+		signal.Stop(notifier)
+
+		interruptMu.Lock()
+		defer interruptMu.Unlock()
```

Comment:
Thought the same. Was hoping someone would suggest to just hold it, too.
Should be fine as long as  it's manually unlocked in the test.